pkg/daemonset/daemonset.go

```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies HostNetwork to daemonset definition.
+func (builder *Builder) WithHostNetwork() *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying hostnetwork flag to daemonset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.HostNetwork = true
+
+	return builder
+}
+
+// WithConfigMapVolume attaches given volume to all pod's containers under daemonset.
+func (builder *Builder) WithConfigMapVolume(volumeName, mountPath string) *Builder {
```

Comment:
Can you align this method with the WithVolume deployment method?
https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L301C25-L301C35

Reply:
For context, volumeMounts should be added through ContainerBuilder - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/pod/container.go#L353

Reply:
I was aligning with "WithLocalVolume" method of pod pkg. But yes, it is more structured to have separate methods for Volume and VolumeMount (container builder). 
I have made changes accordingly. PTAL.

Reply:
Understood, thanks. Will make an issue for refactoring that one